Library/Homebrew/test/cmd/leaves_spec.rb

```diff
@@ -44,4 +44,27 @@
         .and be_a_success
     end
   end
+
+  context "when there are installed formulae with dependencies", :integration_test do
+    it "prints leaves installed as dependencies with --installed-as-dependency" do
+      setup_test_formula "testball"
+      install_test_formula "testball1", 'depends_on "testball"'
+
+      expect { brew "leaves", "--installed-as-dependency" }
+        .to output("testball\n").to_stdout
+        .and not_to_output.to_stderr
+        .and be_a_success
+    end
+  end
```

Comment:
I'm not sure why, but this test doesn't pass. It outputs nothing.

Reply:
We'd probably rather avoid adding more integration tests (they are super slow). If this works for you locally: that's good enough for me (and I'll test it and merge)

Reply:
Fair enough. I've It works for me locally. Let me know if you run into any issues testing it.